_posts/1-4-1-CLI.md

```diff
@@ -6,123 +6,322 @@ section: Extensions
 ---
 
 # Shoutem CLI
-Shoutem Command Line Interface (CLI) is a tool to manage your extensions. By using it you can create extensions, generate code snippets, upload extensions to your Shoutem account and install them on your applications. Here you can find a list of available commands. 
+Shoutem Command Line Interface (CLI) is a tool that helps you build extensions. Using it, you can create extensions, generate code snippets, upload extensions to your Shoutem account and install them on your applications.
 
-
-## Install
-You can download and install Shoutem CLI through `npm`, the package management system for `Node.js`.
+## Installation
+Download and install Shoutem CLI through `npm`, the package manager for `Node.js`.
 
 ```ShellSession
 $ npm install -g @shoutem/cli
 ```
 
 In case you don't have `npm` installed, the best way to install it is with [installing](https://nodejs.org/en/download/) `Node.js`, which includes `npm`.
 
-## Register
-Enter your Shoutem credentials, and register as a developer. If you're not a Shoutem user yet,
-sign up at [www.shoutem.com](https://www.shoutem.com).
+## Commands
+
+Here is the list of available commands.
+
+- ##### shoutem
+
+Lists all the commands.
+
+```ShellSession
+$ shoutem
+Usage: shoutem [command] [-h]
+
+Commands:
+...
+```
+
+- ##### shoutem [\--version|-v]
+
+Prints out the CLI version.
+
+```ShellSession
+$ shoutem -v
+1.0.0
+```
+
+- ##### shoutem \<cmd> [\--help|-h]
+
+Prints help for command.
+
+```ShellSession
+$ shoutem init -h
+
+Usage: shoutem init <extension-name>
+
+Create a scaffold of all files and folders required to build an extension.
+```
+
+- ##### shoutem login
+
+Authenticates you as developer. Enter your Shoutem credentials. If you're not a Shoutem user yet, sign up at [www.shoutem.com](https://www.shoutem.com).
+
+```ShellSession
+$ shoutem login
+Enter your Shoutem credentials:
+E-mail: developer@example.com
+Password: ********
+```
+
+Each Shoutem user can become Shoutem developer. If you still are not a developer, after you entered correct credentials, you'll be prompted a new developer name.
+
+```ShellSession
+Enter new developer name:
+Name: david
+You are registered as `david`!
+```
+
+- ##### shoutem whoami
```

Comment:
Yea, but it's not UX friendly to request to use `login` for this. We need to implement `whoami` (npm has it too).